src/mimswap/periphery/Router.sol

```diff
@@ -119,12 +150,12 @@ contract Router {
             uint256 quoteInputRatio = DecimalMath.divFloor(quoteInAmount, quoteReserve);
             if (baseInputRatio <= quoteInputRatio) {
                 baseAdjustedInAmount = baseInAmount;
-                quoteAdjustedInAmount = DecimalMath.mulFloor(quoteReserve, baseInputRatio);
-                shares = DecimalMath.mulFloor(totalSupply, baseInputRatio);
+                quoteAdjustedInAmount = DecimalMath.mulCeil(quoteReserve, baseInputRatio);
+                shares = DecimalMath.mulCeil(totalSupply, baseInputRatio);
```

Comment:
Ah apologies, shares should still use mulFloor here, it's just the quoteAdjustedInAmount that we should assign to mulCeil in these specific cases

Reply:
Fix in 10089dfece5c127b5c370ea4ac4d86551d691bd0 but not sure if this is exactly what you are suggesting. Would appreciate your confirmation please